Index: freemind/freemind/main/FreeMind.java
===================================================================
--- freemind/freemind/main/FreeMind.java	(.../tags/RELEASE-0-8-0)	(revision 2)
+++ freemind/freemind/main/FreeMind.java	(.../branches/WHFM-0-8-0)	(working copy)
@@ -86,6 +86,7 @@
     private static final String DEFAULT_LANGUAGE = "en";
     private HookFactory nodeHookFactory;
 	public static final String version = "0.8.0";
+    public static final String hackedversion = " (WorldHello.net)";
     //    public static final String defaultPropsURL = "freemind.properties";
     public URL defaultPropsURL;
     //    public static Properties defaultProps;
Index: freemind/freemind/main/FreeMindSplash.java
===================================================================
--- freemind/freemind/main/FreeMindSplash.java	(.../tags/RELEASE-0-8-0)	(revision 2)
+++ freemind/freemind/main/FreeMindSplash.java	(.../branches/WHFM-0-8-0)	(working copy)
@@ -61,6 +61,7 @@
         		g2.setFont(font);
                 // determine width of string to center it.
                 String freemindVersion = frame.getFreemindVersion();
+                freemindVersion += " " + FreeMind.hackedversion;
                 int width = g2.getFontMetrics().stringWidth(freemindVersion);
         		int yCoordinate = (int)(getSize().getHeight())-14;
                 int xCoordinate = (int)(getSize().getWidth()/2-width/2);
Index: freemind/freemind/controller/Controller.java
===================================================================
--- freemind/freemind/controller/Controller.java	(.../tags/RELEASE-0-8-0)	(revision 2)
+++ freemind/freemind/controller/Controller.java	(.../branches/WHFM-0-8-0)	(working copy)
@@ -993,7 +993,7 @@
             this.controller = controller;
         }
         public void actionPerformed(ActionEvent e) {
-           JOptionPane.showMessageDialog(getFrame().getViewport(),controller.getResourceString("about_text")+FreeMind.version);
+           JOptionPane.showMessageDialog(getFrame().getViewport(),controller.getResourceString("about_text")+FreeMind.version + " " + FreeMind.hackedversion);
         }
     }
 
