--- a/freemind/freemind.sh	2006-10-30 04:31:53.000000000 +0800
+++ b/freemind/freemind.sh	2007-03-09 23:57:16.000000000 +0800
@@ -135,6 +135,7 @@
 #--------- Put the environment together --------------------------------
 
 _source /etc/freemind/freemindrc
+_source /opt/freemind/conf/freemindrc
 _source ~/.freemind/freemindrc
 
 findjava
@@ -176,7 +177,10 @@
 	exit 1
 fi
 
-if [ ! -f ~/.freemind/patterns.xml ] && [ -f /etc/freemind/patterns.xml ]
+PATTERNS_XML=/opt/freemind/conf/patterns.xml
+[ -f $PATTERNS_XML ] || PATTERNS_XML=/etc/freemind/patterns.xml
+
+if [ ! -f ~/.freemind/patterns.xml ] && [ -f $PATTERNS_XML ]
 then
 	if [ ! -d ~/.freemind ]
 	then
@@ -184,7 +188,7 @@
 		mkdir -p ~/.freemind
 	fi
 	_debug "Copying patterns.xml to ~/.freemind."
-	cp /etc/freemind/patterns.xml ~/.freemind/patterns.xml
+	cp $PATTERNS_XML ~/.freemind/patterns.xml
 fi
 
 #--------- Call (at last) FreeMind -------------------------------------
@@ -198,6 +202,7 @@
 ${freedir}/lib/bindings.jar:\
 ${freedir}/lib/commons-lang-2.0.jar:\
 ${freedir}/lib/forms-1.0.5.jar:\
+${freedir}/lib/saxon.jar:\
 ${freedir}"
 if [ "${JAVA_TYPE}" = "sun" ]
 then
--- a/freemind/freemind.bat	2007-03-14 15:55:06.000000000 +0800
+++ b/freemind/freemind.bat	2007-03-14 16:03:45.000000000 +0800
@@ -1,2 +1,2 @@
 @echo off
-java -cp lib\freemind.jar;lib\commons-lang-2.0.jar;lib\forms-1.0.5.jar;lib\jibx\jibx-run.jar;lib\jibx\xpp3.jar;lib\bindings.jar freemind.main.FreeMindStarter
+java -cp lib\freemind.jar;lib\commons-lang-2.0.jar;lib\forms-1.0.5.jar;lib\jibx\jibx-run.jar;lib\jibx\xpp3.jar;lib\bindings.jar;lib\saxon.jar freemind.main.FreeMindStarter
